unstructured_inference/models/unstructuredmodel.py

```diff
@@ -55,6 +64,106 @@ def __call__(self, x: Image) -> List[LayoutElement]:
         """Inference using function call interface."""
         return super().__call__(x)
 
+    @staticmethod
+    def deduplicate_detected_elements(
+        elements: List[LayoutElement],
+        min_text_size: int = 15,
+    ) -> List[LayoutElement]:
+        """Deletes overlapping elements in a list of elements. Also will delete elements
+        of less than min_text_size pixels height"""
+
+        if len(elements) <= 1:
+            return elements
+
+        def clean_tables(elements: List[LayoutElement]) -> Iterable[LayoutElement]:
+            import numpy as np
+
+            tables = [e for e in elements if e.type == "Table"]
+            not_tables = [e for e in elements if e.type != "Table"]
+            if len(tables) == 0:
+                return elements
+
+            nested_check = None
+            for table in tables:
+                nested_current = np.array([probably_contained(e, table) for e in not_tables])
+                if nested_check is None:
+                    nested_check = nested_current
+                    continue
+                nested_check = nested_check | nested_current
+
+            final_elements = []
+            for nested, elem in zip(nested_check, not_tables):  # type:ignore
+                if not nested:
+                    final_elements.append(elem)
+
+            final_elements.extend(tables)
+            final_elements.sort(key=lambda e: e.y1)
+            return final_elements
+
+        def enhance_regions(
+            elements: List[LayoutElement],
+            min_text_size: int,
+            iom_to_merge: float = 0.3,
+        ) -> List[LayoutElement]:
+            """This function traverse all the elements deleting nested elements,
+            if are close enough could merge it or split them, depending or the
+            iou score for both regions"""
+            intersections_mtx = intersections(*elements)
+
+            for i, row in enumerate(intersections_mtx):
+                first = elements[i]
+                if first:
+                    # We get only the elements which intersected
+                    indices_to_check = np.where(row)[0]
+                for j in indices_to_check:
+                    if i != j and elements[j] is not None and elements[i] is not None:
+                        second = elements[j]
+                        intersection = first.intersection(
+                            second,
+                        )  # we know it does, but need the region
```

Comment:
so we have computed somewhere -> can we capture that when it was first computed so here we just use the reference instead of recomputing it here?

Reply:
Not currently, but is an issue that could be addressed to get performance improvement 🤔